pinot-common/src/main/java/org/apache/pinot/sql/parsers/rewriter/PredicateComparisonRewriter.java

```diff
@@ -18,21 +18,27 @@
  */
 package org.apache.pinot.sql.parsers.rewriter;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import org.apache.commons.lang3.EnumUtils;
 import org.apache.pinot.common.request.Expression;
+import org.apache.pinot.common.request.ExpressionType;
 import org.apache.pinot.common.request.Function;
 import org.apache.pinot.common.request.PinotQuery;
 import org.apache.pinot.common.utils.request.RequestUtils;
 import org.apache.pinot.pql.parsers.pql2.ast.FilterKind;
 import org.apache.pinot.sql.parsers.SqlCompilationException;
 
-
 public class PredicateComparisonRewriter implements QueryRewriter {
   @Override
   public PinotQuery rewrite(PinotQuery pinotQuery) {
     Expression filterExpression = pinotQuery.getFilterExpression();
     if (filterExpression != null) {
+      filterExpression = updateBooleanPredicates(filterExpression, null);
```

Comment:
I'd do it within the `updateComparisonPredicate` (you may rename it to `updatePredicate`) so that it can be applied to both filter and having, and we just need to traverse the tree once

Reply:
The expression within the WHERE and HAVING clause should be of boolean type, and `sum(field) > 10` will return a boolean result. Some functions can also return boolean result, and is not captured as `FilterKind`, e.g. `isWholeNumber(sum(field))`. For the HAVING clause, the aggregation result is treated as a regular column, so we should handle it the same as WHERE clause.

Reply:
Thanks for the pointer. Handled this case.